src/modules/cpu/kernel/spectrogram.hpp

```diff
@@ -0,0 +1,248 @@
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+#include <chrono>
+#include <complex>
+
+inline Rpp32f reduce_add_ps1(__m256 pSrc)
+{
+    __m256 pSrcAdd = _mm256_add_ps(pSrc, _mm256_permute2f128_ps(pSrc, pSrc, 1));
+    pSrcAdd = _mm256_add_ps(pSrcAdd, _mm256_shuffle_ps(pSrcAdd, pSrcAdd, _MM_SHUFFLE(1, 0, 3, 2)));
+    pSrcAdd = _mm256_add_ps(pSrcAdd, _mm256_shuffle_ps(pSrcAdd, pSrcAdd, _MM_SHUFFLE(2, 3, 0, 1)));
+    Rpp32f *addResult = (Rpp32f *)&pSrcAdd;
+    return addResult[0];
+}
+
+inline void hann_window(Rpp32f *output, Rpp32s windowSize)
+{
+    Rpp32f a = TWO_PI / windowSize;
+    for (Rpp32s t = 0; t < windowSize; t++)
+    {
+        Rpp32f phase = a * (t + 0.5);
+        output[t] = (0.5 * (1.0 - std::cos(phase)));
+    }
+}
+
+inline Rpp32s get_num_windows(Rpp32s length, Rpp32s windowLength, Rpp32s windowStep, bool centerWindows)
+{
+    if (!centerWindows)
+        length -= windowLength;
+    return ((length / windowStep) + 1);
+}
+
+inline Rpp32s get_idx_reflect(Rpp32s idx, Rpp32s lo, Rpp32s hi)
+{
+    if (hi - lo < 2)
+        return hi - 1;
+    for (;;)
+    {
+        if (idx < lo)
+            idx = 2 * lo - idx;
+        else if (idx >= hi)
+            idx = 2 * hi - 2 - idx;
+        else
+            break;
+    }
+    return idx;
+}
+
+RppStatus spectrogram_host_tensor(Rpp32f *srcPtr,
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp32f *dstPtr,
+                                  RpptDescPtr dstDescPtr,
+                                  Rpp32s *srcLengthTensor,
+                                  bool centerWindows,
+                                  bool reflectPadding,
+                                  Rpp32f *windowFunction,
+                                  Rpp32s nfft,
+                                  Rpp32s power,
+                                  Rpp32s windowLength,
+                                  Rpp32s windowStep,
+                                  RpptSpectrogramLayout layout)
+{
+    Rpp32s windowCenterOffset = 0;
+    bool vertical = (layout == RpptSpectrogramLayout::FT);
+    if (centerWindows)
+        windowCenterOffset = windowLength / 2;
+    if (nfft == 0.0f)
+        nfft = windowLength;
+    Rpp32s numBins = nfft / 2 + 1;
+    const Rpp32f mulFactor = TWO_PI / nfft;
+    Rpp32u hStride = dstDescPtr->strides.hStride;
+    Rpp32s alignedNfftLength = nfft & ~7;
+    Rpp32s alignedNbinsLength = numBins & ~7;
+    Rpp32s alignedWindowLength = windowLength & ~7;
+    Rpp32f cosf[numBins * nfft];
+    Rpp32f sinf[numBins * nfft];
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(8)
+    for (Rpp32s k = 0; k < numBins; k++)
+    {
+        Rpp32f* cosfTemp = cosf + (k * nfft);
+        Rpp32f* sinfTemp = sinf + (k * nfft);
+        __m256 pJN = _mm256_set_ps(7,6,5,4,3,2,1,0);
+        __m256 pMulFactorN = _mm256_set1_ps(k*mulFactor);
+        __m256 pAddFactorN = _mm256_set1_ps(8);
```

Comment:
Can use avx_p8 defined rpp_cpu_simd.hpp
https://github.com/sampath1117/rpp/blob/sr/audio_augmentations/src/include/cpu/rpp_cpu_simd.hpp#L99

Instead of using _mm256_set1_ps(8)

Reply:
Done